### PR TITLE
WIP: Git sha1 version fix

### LINKED ISSUE #84: compilation fails due to git build unable to retrieve sha1
While I am not sure if this is an environment specific issue but it seems that I am no longer able to compile the project (gui or console) due to git not being able to retrieve revision. See https://github.com/erikdoe/ccmenu/blob/4f0b1eebb1dca165a3d1aca852be85b9fb98685e/CCMenu.xcodeproj/project.pbxproj#L792

If I comment this line, the build works without any problems. I tried to clean but it does not solve the problem.

What happens:
```
Testing failed:
	Sed: can't read s/#gitsha#//: No such file or directory
** TEST FAILED **
```

Please note that there is no problem with git itself, running `git rev-parse --short HEAD` works very well. My impression is that something went wrong with some folders in xcode, which makes it run the command in the wrong folder.

### PR DESCRIPTION
**Pull request by ssbarnea**
Fixes #84

Change-Id: I51605b1dc9a3759fb5ca2afad22e6e3aca71c907

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 language: objective-c
 osx_image: xcode8.2
 before_install:
-    - gem install xcpretty -N --quiet
+    - curl -sSL https://rvm.io/mpapis.asc | gpg --import -
+    - rvm get stable
```
Comment (erikdoe) [inline .travis.yml]: What is the purpose of this change?

File: CCMenu.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -789,7 +789,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "SHA=`git rev-parse --short HEAD | tr '[a-z]' '[A-Z]'`\nsed -i \"\" \"s/#gitsha#/$SHA/\" \"${BUILT_PRODUCTS_DIR}\"/CCMenu.app/Contents/Info.plist";
+			shellScript = "sed -i\\\"\\\" \"s/#gitsha#/`git rev-parse --short HEAD | tr '[a-z]' '[A-Z]'`/\" \"${BUILT_PRODUCTS_DIR}/CCMenu.app/Contents/Info.plist\"";
```
Comment (erikdoe) [inline CCMenu.xcodeproj/project.pbxproj]: So, this change basically inlines the variable. As far as I can see there are no other changes. I get that the previous version doesn't work in your environment, but it does work elsewhere. Can I ask what shell you are using and whether you have added some "special" configuration to your shell? Because, otherwise, why would the variable assignment work in most cases, but not in your environment?

File: .travis.yml (hunk 1 above)
Comment (ssbarnea) [reply .travis.yml]: @erikdoe Indeed very valid question. Here is how travis execution was looking before installing the gpg key https://travis-ci.org/erikdoe/ccmenu/builds/327436075#L385

I moved the installation of xcpretty into the makefile to make it easier for the developer to build. 

I am not sure if the rvm part should be in travis file or in makefile, I had a lot of trouble installing rvm on my machine in order be able to build and having code in travis.file does not help developers.

I didn't had time to finish the changes but I think that you already figured out that my goal was to enable anyone to build ccmenu after checking out the code. There were few hardcoded stuff that request changing the source code in order to build or to run tests.

I am curious what is you opinion regarding best way to address this. I don't want to spend too much time in something that will not endup as being used.

**Comment by erikdoe** [comment]
I don't think we ever got to the bottom of why the implementation in the script doesn't work for you (see my question further up). Did you make any progress here or should I close this PR?
